packages/sewing-kit-koa/src/assets.ts

```diff
@@ -43,21 +55,61 @@ export default class Assets {
   }
 
   async styles({name = 'main'} = {}) {
-    const {css} = getAssetsForEntrypoint(name, await this.getAssetList());
+    const {css} = getAssetsForEntrypoint(
+      name,
+      await this.getResolvedManifest(),
+    );
     return css;
   }
 
-  private async getAssetList() {
-    if (this.resolvedAssetList) {
-      return this.resolvedAssetList;
+  private async getResolvedManifest() {
+    if (this.resolvedManifest) {
+      return this.resolvedManifest;
+    }
+
+    const consolidatedManifest = await loadConsolidatedManifest();
+
+    if (consolidatedManifest.length === 0) {
+      throw new Error('No builds were found.');
     }
 
-    this.resolvedAssetList = await loadConsolidatedManifest();
-    return this.resolvedAssetList;
+    const {userAgent} = this;
+
+    // We do the following to determine the correct manifest to use:
+    //
+    // 1. If there is no user agent, use the "last" manifest, which is the
+    // least restrictive set of browsers.
+    // 2. If there is only one manifest, use it, regardless of how well it
+    // matches the user agent.
+    // 3. If there is a user agent, find the first manifest where the
+    // browsers it was compiled for matches the user agent, or where there
+    // is no browser restriction on the bundle.
+    // 4. If no matching manifests are found, fall back to the last manifest.
+    if (userAgent == null) {
+      this.resolvedManifest =
+        consolidatedManifest[consolidatedManifest.length - 1].manifest;
```

Comment:
can we pull `consolidatedManifest[consolidatedManifest.length -1]` into a semantically named variable?